### PR TITLE
[Bug](bloomfilter) Fix bloom filter for date type

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

image

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Gabriel39** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.07 seconds
 stream load tsv:          449 seconds loaded 74807831229 Bytes, about 158 MB/s
 stream load json:         25 seconds loaded 2358488459 Bytes, about 89 MB/s
 stream load orc:          72 seconds loaded 1101869774 Bytes, about 14 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230329074716_clickbench_pr_122190.html

File: be/src/exprs/bloom_filter_func.h [hunk 1]
```diff
@@ -330,6 +330,11 @@ struct DateFindOp : public CommonFindOp<vectorized::VecDateTimeValue> {
 
         vectorized::VecDateTimeValue date_value;
         date_value.from_olap_date(value);
+        // So confusing here. For join node with condition (a.date_col = b.date_col), the actual
+        // expression is CAST(a.date_col AS DATETIME) = CAST(b.date_col AS DATETIME). So we build
+        // this bloom filter by CAST(a.date_col AS DATETIME) and also need to probe this bloom
+        // filter by a datetime value.
+        date_value.set_type(3);
```
Comment (yiguolei) [inline be/src/exprs/bloom_filter_func.h:337]: set_type(TimeType::TIME_DATETIME)

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**